docs/data/data-grid/recipes-editing/recipes-editing.md

```diff
@@ -105,3 +105,14 @@ Using the [controlled mode](/x/react-data-grid/editing/#controlled-model) and li
 The following demo implements this behavior.
 
 {{"demo": "SingleClickEditing.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Bulk editing
+
+The data grid [Editing](/x/react-data-grid/editing/) API exposes [the `processRowUpdate` callback](/x/react-data-grid/editing/#the-processrowupdate-callback) which is commonly used to persist edits on per-row basis.
+You can utilize this callback to batch edits locally and then choose to either persist or discard them in bulk.
```

Comment:
**Off-topic to this PR:**
Should we consider supporting bulk-editing internally in the future?

I agree that with the client-side editing, it may not look worth the effort but would it make sense with the new data source implementation that is supposed to handle editing as well, the currently proposed [`dataSource.updateRow`](https://www.notion.so/mui-org/Data-source-draft-db58b8b749154420a407f231c5528c1d?pvs=4#08c12d2ddfea46f6b83d915c0084b35e) callback will only handle single-row editing and to support bulk-editing on the server-side, the users will have to fallback to a solution similar to the one in this PR.

Probably adding an internal state to track which rows have unsaved (or uncommitted) changes and allowing users to mark them as saved at will might be a good direction to proceed in.

Wdyt?

Reply:
> Should we consider supporting bulk-editing internally in the future?

In the future - maybe, but I wouldn't increase the scope of the Data Source now.
Also, if we can make a React hook for bulk editing instead, it could be even better